src/main/java/htsjdk/samtools/LinearBAMIndex.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+package htsjdk.samtools;
+
+import htsjdk.samtools.CachingBAMFileIndex;
+import htsjdk.samtools.LinearIndex;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.seekablestream.SeekableStream;
+
+/**
+ * The htsjdk APIs for accessing the linear BAM index are private...
+ */
+public class LinearBAMIndex extends CachingBAMFileIndex {
```

Comment:
Hi @tomwhite! I've made this public because it is located in `htsjdk.samtools` and needs to be accessed from `org.seqdoop.hadoop_bam`. The class has to be in `htsjdk.samtools` because it needs access to the package-private `CachingBAMFileIndex` class.

Reply:
Oh yes, this is in a different package so it needs to be public. Can you open an htsjdk issue to increase its visibility please?

Reply:
Definitely, will do.

Reply:
Opened https://github.com/samtools/htsjdk/issues/975.